src/lib/dialog/dialog.ts

```diff
@@ -143,10 +174,16 @@ export class MdDialog {
     let userInjector = config && config.viewContainerRef && config.viewContainerRef.injector;
     let dialogInjector = new DialogInjector(dialogRef, userInjector || this._injector);
 
-    let contentPortal = new ComponentPortal(component, null, dialogInjector);
+    if (component) {
```

Comment:
I think that the logic that is specific to `TemplateRef` and `ComponentType` should be split into separate private methods.